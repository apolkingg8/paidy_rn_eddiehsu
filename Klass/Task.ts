import {v4} from "uuid";

export default class Task {
    id: string = v4()
    text: string = ''

    constructor(props: Partial<Task>) {
        Object.assign(this, props)
    }
}
