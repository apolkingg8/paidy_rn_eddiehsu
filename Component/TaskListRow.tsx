import {Pressable, StyleSheet, Text, View} from "react-native";
import {observer} from "mobx-react";
import Task from "../Klass/Task";
import styleStore from "../store/styleStore";
import {percent} from "csx";
import taskStore from "../store/taskStore";

interface TaskRowProps {
    task: Task
}

const TaskListRow = (props: TaskRowProps)=> {
    return (
        <View style={styles.wrap}>
            <View style={styles.textWrap}>
                <Text style={styles.text}>
                    {props.task.text}
                </Text>
            </View>
            <Pressable
                style={styles.button}
                onPress={()=> {
                    taskStore.removeTaskById(props.task.id)
                }}
            >
                <Text style={styles.buttonText}>âœ•</Text>
            </Pressable>
        </View>
    )
}

const styles = StyleSheet.create({
    wrap: {
        ...styleStore.centerRow,
        width: percent(100),
        minHeight: 40,
        marginVertical: 8,
        paddingHorizontal: 24,
    },
    textWrap: {
        ...styleStore.centerRow,
        justifyContent: "flex-start",
        flex: 1,
        height: percent(100),
    },
    text: {
        color: styleStore.palette.white,
    },
    button: {
        ...styleStore.centerRow,
        width: 40,
        height: percent(100),
    },
    buttonText: {
        color: styleStore.palette.primary,
    },
})

export default observer(TaskListRow)
