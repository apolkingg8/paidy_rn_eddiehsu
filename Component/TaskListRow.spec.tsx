import Task from "../Klass/Task";
import {render} from "@testing-library/react-native";
import TaskListRow from "./TaskListRow";

describe('TaskListRow', ()=> {
    test('match snapshot', ()=> {
        const task1 = new Task({id: 'task1', text: 'Task1'})
        const component = render(<TaskListRow task={task1}/>).toJSON()
        expect(component).toMatchSnapshot()
    })
})
