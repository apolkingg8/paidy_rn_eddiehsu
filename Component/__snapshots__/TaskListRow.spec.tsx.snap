// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskListRow match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 8,
      "minHeight": 40,
      "paddingHorizontal": 24,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#f7f7f7",
        }
      }
    >
      Task1
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "paddingHorizontal": 24,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#b0e55a",
        }
      }
    >
      âœ•
    </Text>
  </View>
</View>
`;
